apps/src/templates/progress/BubbleFactory.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
```

Comment:
I think the functionality here ended up being less encapsulated than I imagined in my head -- this feels more like a set of helper functions than a factory pattern that abstracts the creation of bubble components.  I don't have a clear picture of how the other bubbles are the same/different as the ones in progress table so I'm curious to see how this evolves as we bring in the other bubbles.

Reply:
i agree. this was just a quick pass at moving common bubble stuff out of `ProgressTableLevelBubble`, which had accumulated a decent number of internal helpers. so much of our bubble code is devoted to translating props (e.g. bubble should be disabled if `isDisabled || status === locked`), so i may end up moving `BasicBubble` into its own module and replacing `BubbleFactory` with `bubbleHelpers` or something like that. i might like to schedule a call with you and Maureen to discuss options as i develop a clearer picture of the varying needs between bubble use cases.